dbt/include/azuresynapse/macros/materializations/snapshot/strategy.sql

```diff
@@ -1,4 +1,4 @@
-{% macro snapshot_check_strategy(node, snapshotted_rel, current_rel, config, target_exists) %}
+{% macro azuresynapse__snapshot_check_strategy(node, snapshotted_rel, current_rel, config, target_exists) %}
```

Comment:
Hi @mattberns I have named the macro to override [`snapshot_check_strategy` in the core](https://github.com/fishtown-analytics/dbt/blob/daff0badc82362d805bce0a2cc8a59701ab1fefb/core/dbt/include/global_project/macros/materializations/snapshot/strategies.sql#L131). This macro is called [here](https://github.com/fishtown-analytics/dbt/blob/daff0badc82362d805bce0a2cc8a59701ab1fefb/core/dbt/include/global_project/macros/materializations/snapshot/snapshot.sql#L179) without `adapter.dispatch`, so renaming the macro breaks the check strategy.

Reply:
@seunghanhong  For some reason, when I was working with this locally, I was getting an error invalid duplicate macro:

    Compilation Error
      dbt found two macros with the name "snapshot_check_strategy" in the namespace "dbt". Since these macros have the same name and exist in the same namespace, dbt will be unable to decide which to call. To fix this, change the name of one of these macros:
      - macro.dbt_azuresynapse.snapshot_check_strategy (macros/materializations/snapshot/strategy.sql)
      - macro.dbt_azuresynapse.snapshot_check_strategy (macros/materializations/snapshot/strategy.sql)

I have done some more research on my side and now I think I probably just had some invalid setup or something causing this, changing this back runs swimmingly.

My bad, hotfix PR inbound!